src/RedemptionPool.sol

```diff
@@ -98,74 +128,88 @@ contract RedemptionPool is Ownable {
-        _userBalance[msg.sender] -= _amount;
-        totalGRO -= _amount;
-        GRO.safeTransfer(msg.sender, _amount);
-        emit Withdraw(msg.sender, _amount);
-    }
-
-    /// @notice Allow to withdraw cUSDC based on amount of GRO tokens deposited per user
-    function claim() external {
-        if (block.timestamp <= DEADLINE) {
-            revert RedemptionErrors.DeadlineExceeded();
-        }
-        if (_userBalance[msg.sender] <= 0) {
+    /**
+     * @notice Allow users to claim their share of USDC tokens based on the amount of GRO tokens they have deposited
+     * @dev Users must have a positive GRO token balance and a non-zero claim available to make a claim
+     * @dev The deadline for making GRO deposits must have passed
+     * @dev Redeems the user's cUSDC tokens for an equivalent amount of USDC tokens and transfers them to the user's address
+     * @dev Decreases the user's claims and contract accounting by the amount claimed
+     */
+    function claim() external onlyAfterDeadline {
+        // Check if the user has a positive GRO token balance
+        if (_userGROBalance[msg.sender] <= 0)
             revert RedemptionErrors.NoUserBalance();
-        }
+
+        // Get the amount of cUSDC tokens available for the user to claim
         uint256 userClaim = getSharesAvailable(msg.sender);
-        if (userClaim == 0) {
-            revert RedemptionErrors.NoUserClaim();
-        }
+
+        // Check if the user has a non-zero claim available
+        if (userClaim <= 0) revert RedemptionErrors.NoUserClaim();
+
+        // Redeem the user's cUSDC tokens for USDC tokens
+        // and transfer the USDC tokens to the user's address
+        uint256 usdcAmount = ICERC20(CUSDC).redeem(userClaim);
+        IERC20(USDC).safeTransfer(msg.sender, usdcAmount);
+
+        // Decrease the user's claims and contract accounting by the amount claimed
         _userClaims[msg.sender] += userClaim;
-        IERC20(CUSDC).transfer(msg.sender, userClaim);
+        totalCUSDCWithdrawn += userClaim;
         emit Claim(msg.sender, userClaim);
     }
 
     /////////////////////////////////////////////////////////////////////////////
     //                              Permissioned funcs                         //
     /////////////////////////////////////////////////////////////////////////////
 
-    /// @notice Pulls assets from the msig
-    /// @param amount amount of cUSDC to pull
-    function depositCUSDC(uint256 amount) external onlyOwner {
-        require(amount > 0, "amount must be greater than 0");
-        require(IERC20(CUSDC).transferFrom(msg.sender, address(this), amount), "transfer failed");
-        totalAssetsDeposited += amount;
-        emit CUSDCDeposit(totalAssetsDeposited);
+    /// @notice Pulls assets from the DAO msig
+    /// @param _amount amount of cUSDC to pull
+    function depositCUSDC(uint256 _amount) external onlyOwner {
+        if (_amount <= 0) revert RedemptionErrors.GreaterThanZeroOnly();
```

Comment:
Cannot be negative

Reply:
Okeo. Just replicating old logic but can simplify with = operator instead